src/scheduler/mod.rs

```diff
@@ -183,16 +197,25 @@ impl PerCoreScheduler {
 
 		// Add it to the task lists.
 		let wakeup = {
+			#[cfg(feature = "smp")]
 			let mut input_locked = get_scheduler(core_id).input.lock();
 			TASKS.lock().insert(tid, VecDeque::with_capacity(1));
 			NO_TASKS.fetch_add(1, Ordering::SeqCst);
+			#[cfg(feature = "smp")]
 			if core_id != core_scheduler().core_id {
 				input_locked.new_tasks.push_back(clone_task);
 				true
 			} else {
 				core_scheduler().ready_queue.push(clone_task);
 				false
 			}
+			#[cfg(not(feature = "smp"))]
+			if core_id == core_scheduler().core_id {
```

Comment:
Is the core_id a constant when smp is disabled?

Reply:
core_id is constant.

Reply:
Can you turn `core_scheduler().core_id` into a compile time constant?

Reply:
No, but I am able to remove it.